presto-product-tests-launcher/src/main/java/io/prestosql/tests/product/launcher/env/Environment.java

```diff
@@ -150,6 +157,11 @@ private Environment tryStart()
 
     public void stop()
     {
+        if (attached) {
+            // do not stop attached, externally controlled environment
```

Comment:
The fact that stop does not stop is not intuitive.

perhaps, make it `throw illegal state` and update  `test attach` not to call stop, if there is nothing to stop.
or, just change the `test attach` flow and then no changes needed in Env clas


also, can the test container leak in `test attach` mode? 
(i hope it's controller by ryuk and "safe to leak")

Reply:
> The fact that stop does not stop is not intuitive.

Fact. I will move this `if` to `close` method and throw in  `stop`

> also, can the test container leak in test attach mode?

I tested it and it was cleared. I will test it more with interruptions.

Reply:
> I tested it and it was cleared. I will test it more with interruptions.

It works as expected.